### PR TITLE
aims_path rework for task-farm support

### PR DESCRIPTION
**Pull request by ikowalec**

### CODE REVIEW

File: carmm/run/aims_path.py [hunk 1]
```diff
@@ -19,36 +19,40 @@ def set_aims_command(hpc='hawk', basis_set='light', defaults=2010, nodes_per_ins
...
-    os.environ["ASE_AIMS_COMMAND"]= preamble + fhi_aims_directory + executable
-    os.environ["AIMS_SPECIES_DIR"] = fhi_aims_directory + species
+    preamble = preamble_types[hpc]
+    fhi_aims_directory = fhi_aims_directories[hpc]
 
+    if nodes_per_instance:
+        task_farmed_commands = {
+            "archer2": "--nodes=" + str(nodes_per_instance) + " --ntasks=" + str(int(128 * nodes_per_instance)) + " ",
+            "hawk": "--nodes=" + str(nodes_per_instance) + " --ntasks=" + str(int(40 * nodes_per_instance)) + " ",
+            # TODO: add and test isambard and young task-farmed commands
```
Comment (logsdail) [inline carmm/run/aims_path.py:50]: If you're not going to implement these outright, I'd suggest putting an Exception here stating not implemented/tested. If you are going to test, all good to leave as is here (i.e. shouldn't leave possibility of buggy behaviour for some future user to find)

Comment (logsdail) [reply carmm/run/aims_path.py:50]: (With reference to Isambard/Young)

File: carmm/run/aims_path.py [hunk 2]
```diff
@@ -19,36 +19,40 @@ def set_aims_command(hpc='hawk', basis_set='light', defaults=2010, nodes_per_ins
...
+            "isambard": "",
+            "young": "",
+        }
+
+        preamble += task_farmed_commands[hpc]
+
+    os.environ["ASE_AIMS_COMMAND"] = preamble + fhi_aims_directory + executable
```
Comment (logsdail) [inline carmm/run/aims_path.py:59]: I'm very happy with the condensation into dictionaries. TBH, I think you can just call the dictionaries `preamble` and `fhi_aims_directory` (I prefer these names, very clear on what they are!), delete L43/44 and then reference above/below to `preamble[hpc]` and `fhi_aims_directory[hpc]`.

File: carmm/run/aims_path.py [hunk 3]
```diff
@@ -19,36 +19,40 @@ def set_aims_command(hpc='hawk', basis_set='light', defaults=2010, nodes_per_ins
...
-        preamble = gerun
-    else:
-        raise Exception("Inappropriate HPC facility: " + hpc + "is not recognised")
-
-    os.environ["ASE_AIMS_COMMAND"]= preamble + fhi_aims_directory + executable
-    os.environ["AIMS_SPECIES_DIR"] = fhi_aims_directory + species
+    preamble = preamble_types[hpc]
+    fhi_aims_directory = fhi_aims_directories[hpc]
 
+    if nodes_per_instance:
```
Comment (logsdail) [inline carmm/run/aims_path.py:46]: This doesn't need to be in an `if` statement - just define irrespective.

Instead put the if around L57 i.e. 
if nodes_per_instance:
ASE_AIMS_COMMAND = .... + task_farmed_commands[hpc] + ....
else:
ASE_AIMS_COMMAND = ....

**Comment by logsdail** [comment]
In principle no problem with the changes - just picked out how the dictionaries can be used at the command definition, rather than in pre-construction, and suggest we just make that minor adjustment?

File: carmm/run/aims_path.py (hunk 3 above)
Comment (logsdail) [reply carmm/run/aims_path.py:44]: Clearly I got that one wrong! All good if resolved.

**Comment by logsdail** [comment]
Assuming this still works for you, it's good to merge.

**Comment by ikowalec** [comment]
> Assuming this still works for you, it's good to merge.

Cool, I will squash the commits and merge.

**Pull request merged**